src/chains/ethereum/ethereum/src/miner/miner.ts

```diff
@@ -500,7 +500,14 @@ export default class Miner extends Emittery<{
 
   #reset = () => {
     this.#origins.clear();
-    this.#priced.clear();
+    const priced = this.#priced;
```

Comment:
This is a slightly hacky fix.

When the priced heap is `reset`, meaning we're clearing out the heap
and origins list to be set again when the miner is called, loop over
the priced heap transactions and "unlock" them (set `tx.locked = false`)

The real fix would include fixing the use of `locked`, as it's
currently overloaded to mean "is in priced heap" and also "is being
mined".